### ISSUE TITLE
Geologic Unit Dataset

### ISSUE THREAD

**Issue by hazardgoat** [GenericMappingTools/pygmt#1364]
**Description of the desired feature**
I would like a geologic unit dataset to be added to pygmt.datasets. After receiving help on the forums for [plotting a geologic map](https://forum.generic-mapping-tools.org/t/pygmt-coloring-polygons-from-a-converted-kml-file/1798), I'd like to contribute what I learned to the example gallery for how to use fig.plot to color polygons by the values in a column. However the PyGMT Contributors Guide says that examples should use data from pygmt.datasets.

**Are you willing to help implement and maintain this feature?** Yes/No
 I am willing to contribute whatever I'm able to bringing this about and maintaining it across new PyGMT releases, though I admit my coding experience is limited.

**Comment by meghanrjones** [comment]
Thanks for opening this issue. I agree that this would be a helpful example. Do you have a dataset in mind for the example that you would like added to pygmt.datasets? Ideal datasets for examples are relatively small and open source.

**Comment by hazardgoat** [comment]
@meghanrjones The dataset I had in mind is a geologic map of the US state of California, provided by the California Geological Survey [here](https://www.conservation.ca.gov/cgs/Documents/Publications/Geologic-Data-Maps/GDM_002_GMC_750k_v2_GIS.zip). As it is a map provided by the state government for the public, I should think that it is freely available for non commercial use. The polygon file itself  is ~47.6MB after conversion to .gmt. I think this would be a good dataset as it is in the vein of what would used in a professional project.

**Comment by weiji14** [comment]
Similar to https://github.com/GenericMappingTools/gmtserver-admin/issues/100, should we move this issue to https://github.com/GenericMappingTools/gmtserver-admin @meghanrjones? Since this would be of benefit to GMT users as well.

**Comment by seisman** [comment]
I don't think this geological unit dataset should be added to the GMT server, because this dataset only provides data for a small region (California in the US), while all other GMT datasets have global data coverage.

@hazardgoat's geologic map is a great example and definitely should be added to the gallery, but it would be better if a smaller dataset can be used.
